### DISCUSSION TITLE
ESP32-S3 Display Module ESP32-8048S043

### DISCUSSION THREAD

**Discussion by 001oc** [arendst/Tasmota#20754] [Q&A]
Hi there,

can someone help me out with some init files (display.ini, init.bat, autoexec.be, autoconfig, ...) for the ESP32-S3-8048S043, 4,3", 800x480 Display? I have got mine from aliexpress with, of course, no docs. As of that I am looking for a starting point to get the tisplay and the touch working with tasmota.

Thanks in advance

Michael

**Comment by 001oc** [comment]
When I upload from https://nightly.openhasp.com/ a bin file for:

Sunton 8048S043C, 16 MB4 MB, ILI9341, XPT2406 I get a working setup with correct display output and working touchscreen.

What I think could be taken from that concerning my question, the display/touch combination seams to be a 4,3" ILI9341 display and XPT2406 touch with size of 800x480.

Can someone give me a set of example files to get this combination working to start over with tasmota?

Thanks

**Reply by 3xtant** [reply]
@001oc I am a novice myself, but I got my ILI9341 with the -lvgl image using universal display and the display.ini file from the repo. That should get the screen working for you.
I have a similar issue with a new touch chip and I'll be watching this thread to see if I follow similar steps.

**Reply by sfromis** [reply]
A recent development is an [Universal Touch Driver](https://tasmota.github.io/docs/Displays/#universal-touch-driver) which "should" allow specifications in the `display.ini` file to match different touch controller chips, without these already being known to the Tasmota core code. As the examples suggests, creating the config for each chip does take some detailed knowledge of how the chip operates.

**Reply by sfromis** [reply]
When you above write "XPT2406", I was wondering if you instead mean the pretty common "XPT2046" instead. Both ILI9341 and XPT2046 are already supported by Tasmota, but to use the touch you need to config and calibrate it in the `display.ini` file.

**Reply by 001oc** [reply]
@3xtant thanks for your reply. I think I have tried it that way, my display.ini looks like that:

```
:H,ILI9341,800,480,16,RGB,40,41,39,42,2,15,16,4,45,48,47,21,14,8,3,46,9,1,5,6,7,14
:S,2,1,1,0,40,20
:V,0,8,4,8,0,8,4,8,1
:0,00
:1,01
:2,02
:3,03
:TI1,5d,*,*,-1,38
#
```
But maybe there are some failures within the file - are you willing to share your display.ini with me to give it a try?

Whats different is, that I have not used the precompiled -lvgl but a selfcompiled version that supports my ESP32-S3 (which wasn't the case with the precompiled -lvgl version).

@sfromis thanks to you too, maybe you are right. I am not able to check wether it is the XPT2406 or the XPT2046 right now. I have used the tool at: [](https://nightly.openhasp.com/)

![grafik](https://github.com/arendst/Tasmota/assets/86658335/10251eed-ad1d-452b-8a13-012a55ceb833)

There they write alot XPT2406 but no XPT2046 - but that is maybe a typo, I don't know. But when using the marked binary screen and even touch are working.

**Reply by sfromis** [reply]
It seems that Sunton has a lot of display variants with different display controllers and touch types/chips, meaning that it is not very useful to make an even slightly imprecise pick when looking for specs. And the openHASP site may not clarify things all that much..... Since XPT2046 is not there at all, I'd assume that XPT2406 is indeed a typo.

Ax XPT2046 is for resistive touch, and the mentioned "Sunton 4827S043C" has a "C" implying (?) capacitive instead of resitive, I'm not very convinced about it having XPT2046 either.... Of course, I'm not all that sure that the "C" *actually* means capacitive, either. Sunton is a mess.

**Reply by sfromis** [reply]
If you're seeing two flat ribbon cables connecting the display to the main PCB, you may have the touch controller chip on or near the smaller of those ribbons. XPT2046 is a 16-pin IC with 8 pins on two sides. On a 7" Sunton display with capacitive touch, I'm seeing a GT911 with about 13 pins on each of 4 sides.

**Reply by 001oc** [reply]
@sfromis ... and now it becomes strange. There are the 2 ribbon cables one larger and one smaller ones. At the smaller one there is a chip soldered on top of the ribbon cable that shows GT911 :-). Nevertheless when I use the bin file as described above for the "XPT2046" touchscreen it is working - very strange.

**Reply by 001oc** [reply]
![20240219_001435](https://github.com/arendst/Tasmota/assets/86658335/0d6bfca7-7598-4b74-b138-ea0ab3f1fa78)
![20240219_001452](https://github.com/arendst/Tasmota/assets/86658335/f9b8f696-8dfe-4bd8-8e92-0acf2e45c641)

**Reply by sfromis** [reply]
As the GT911 is connected via the I2C bus, it is very probable that openHASP can detect the chip, and then automatically select the driver, no matter if XPT2046 (which needs a special pin config) was implied by the defaults in the binary.

On my board, the GT911 works with this in `display.ini`: `:TI1,5d,*,*,-1,38` where 1 is bus, 5d is the address, next two are I2C pins from the template (gpio 20/19), then IRQ not used and RST on pin 38. Possibly your board may not use use same pins.

**Reply by 001oc** [reply]
@sfromis thanks alot, I will give it a try and come back with the result tommorow.

**Reply by 3xtant** [reply]
@001oc I am using the "stock" display.ini file. You can find it in your repo in the `tasmota/displaydesc` folder, or online here: https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/ILI9341_display.ini

I think, though, that since your screen is bigger, you want to keep the customization you made in the first line. It also looks like you are providing the GPIOs in the config rather than via the tasmota config, which is fine if you prefer that.

**Reply by sfromis** [reply]
BTW. That U1 on your board image (near the SD card slot) looks a lot like it could be XPT2046, but it is rather plausible that same board is used both with resistive touch (XPT2046) and the better capacitive touch (GT911), with different ways connecting the two touch layer types.

On my board, the very similar U1 position is unpopulated, but the board is designed to accommodate it. Both boards are pretty similar, but many things are not placed the same way.

**Reply by sfromis** [reply]
@3xtant When the connection type is RGB, the 16 pins cannot be supplied in the template, but has to be in the `display.ini` file. But ILI9341 is usually SPI, where my preference is to have the SPI pins in the normal template.

**Comment by 001oc** [comment]
I have tried all of the above suggestions but couldn't fix my problem.

As of that I started over and installed a precompiled ESP32-S3 basic-binary, after that I made an ota-update with [tasmota32s3-lvgl.bin] from (https://github.com/Jason2866/Tasmota-specials/blob/firmware/firmware/tasmota32/other/tasmota32s3-lvgl.bin) . After that I have done an autoconfig with SunTon-S3-Display-5Inch. That of cause installed an autoconf-file which has some binary values in it and isn't supposed to be edited - at least I think so. There is a display.ini too which has even (different) values to config the display and the touchscreen. When I modify the display.ini it seams to have no influence, at least the display still is not working. As of that I reverted the autoconfig which purged the autoconfigfile too. After that solely the display.ini file is existing. The display is still not working. The content of the display.ini file currently is as follows:

```
:H,ILI9341,800,480,16,RGB,40,41,39,42,2,15,16,4,45,48,47,21,14,8,3,46,9,1,5,6,7,14
:S,2,1,1,0,40,20
:V,0,8,4,8,0,8,4,8,1
:0,00
:1,01
:2,02
:3,03
:TI1,5d,*,*,-1,38
#

```

Hier noch ein Bild von der Rückseite des verwendeten Displays:

![20240219_070741](https://github.com/arendst/Tasmota/assets/86658335/f54c5022-211d-4a0a-a42e-2afa0f3bc6d4)

**Reply by 001oc** [reply]
by the way, that is what comes up at the serialport when starting with the working binary from openHASP

```
--- Terminal on COM6 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x128
load:0x403c9700,len:0xb90
load:0x403cc700,len:0x28cc
entry 0x403c98b8



        open____ _____ _____ _____
          |  |  |  _  |   __|  _  |
          |     |     |__   |   __|
          |__|__|__|__|_____|__|
        Home Automation Switch Plate
        Open Hardware edition v0.7.0-rc10

[       0.105][212980/246736 13][    0/    0  0] DBUG: Started @ 115200 bps
␛[1000D␛[0K#␛[1000D␛[0C[       0.113][212980/246736 13][    0/    0  0] DBUG: Environment: sunton-8048s043c_16MB
␛[1000D␛[0K#␛[1000D␛[0C[       0.123][212980/246736 13][    0/    0  0] UART: Started
␛[1000D␛[0K#␛[1000D␛[0C[       0.131][212980/246736 13][    0/    0  0] UART: Client login from serial
␛[1000D␛[0K#␛[1000D␛[0C[       0.140][212980/246736 13][    0/    0  0] CONF: SPI flash FS mounted
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.163][212980/246736 13][    0/    0  0] CONF: Loading /config.json
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.173][212980/245904 13][    0/    0  0] CONF: Loaded /config.json
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.183][212980/246800 13][    0/    0  0] DBUG: Loading debug settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.192][212980/246800 13][    0/    0  0] GPIO: Loading GUI settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.202][212980/246800 13][    0/    0  0] GUI : First Touch Calibration enabled
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.213][212980/246800 13][    0/    0  0] HASP: Loading HASP settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.222][212980/246800 13][    0/    0  0] WIFI: Loading WiFi settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.252][212980/244832 13][    0/    0  0] WG  : Loading WireGuard settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.262][212980/244832 13][    0/    0  0] MQTT: Loading MQTT settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.279][212980/244832 13][    0/    0  0] TELN: Loading Telnet settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.289][212980/244832 13][    0/    0  0] MDNS: Loading MDNS settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.299][212980/244832 13][    0/    0  0] HTTP: Loading HTTP settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.309][212980/244832 13][    0/    0  0] GPIO: Loading GPIO settings
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.319][212980/244832 13][    0/    0  0] CONF: Settings loaded
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.379][204788/215980  5][    0/    0  0] WIFI: Connecting to XXXXXXX
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.388][204788/215768  5][    0/    0  0] WIFI: auth failed
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.398][204788/215532  4][    0/    0  0] MSGR: Starting...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.407][204788/215840  5][    0/    0  0] MSGR: Started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.415][204788/215840  5][    0/    0  0] TFT : Starting...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.424][204788/215764  5][    0/    0  0] TFT : Starting...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.429][204788/215508  4][    0/    0  0] WIFI: Connected to XXXXXXX, requesting IP...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.448][196596/212760  7][    0/    0  0] WIFI: Received IP address 192.168.2.183
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.479][204788/212828  3][    0/    0  0] TFT : Started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.519][204788/212828  3][    0/    0  0] TFT : Started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.616][204788/212828  3][    0/    0  0] LVGL: Starting...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.629][126964/135812  6][62608/62832  1] LVGL: LV_FS_PC ready
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.638][126964/135812  6][62608/62824  1] LVGL: /littlefs/config.json
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[       0.652][102388/110640  7][61908/61956  1] GUI : Initialize Cursor
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.782][98292/109832 10][61816/61840  1] DRVR: GT911 Started (480x272)
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.796][98292/109832 10][61816/61840  1] DRVR:     *  Found device 0x5d
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.809][98292/109732 10][61576/61592  1] LVGL: Started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.826][98292/109492 10][59648/59764  1] HASP: Custom theme loaded
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.849][98292/109492 10][58544/58584  1] HASP: Loading /pages.jsonl
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.875][98292/109492 10][58452/58464  1] HASP: Loaded /pages.jsonl
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.885][98292/109492 10][58452/58464  1] HASP: Changing page to 1
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.894][98292/109492 10][58292/58304  1] GPIO: Starting...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.903][98292/109492 10][58292/58304  1] GPIO: Reboot counter 0
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.912][98292/109492 10][58292/58304  1] GPIO: Started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.922][94196/104916 10][58292/58304  1] HTTP: Started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.931][94196/104348  9][58292/58304  1] TELN: Telnet console started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.941][94196/104348  9][58292/58304  1] FTP : Starting...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.950][94196/104348  9][58292/58304  1] FTP : Disabled
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.980][94196/104348  9][58292/58304  1] MSGR: File not found: L:/boot.cmd
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:46.998][94196/104348  9][58292/58304  1] MSGR: Loading L:/online.cmd
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.020][86004/98268 12][58292/58304  1] MSGR: jsonl={"page":0,"id":239,"obj":"msgbox","text":"%ip%","auto_close":20000}
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.068][90100/103276 12][57700/57728  1] MSGR: Loaded L:/online.cmd
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.078][94196/103880  9][57700/57728  1] HTTP: Started @ http://192.168.2.183
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.089][94196/103928  9][57700/57728  1] MDNS: Starting...
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.119][86004/97480 11][57700/57728  1] MDNS: Started
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.130][86004/96532 10][57700/57728  1] MQTT: hasp/plate/LWT
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.139][86004/96484 10][57700/57728  1] MQTT: hasp/plate/LWT
␛[1000D␛[0KPrompt > ␛[1000D␛[9C[08:44:47.148][86004/96628 10][57700/57728  1] MQTT: Broker not configured
␛[1000D␛[0KPrompt > ␛[1000D␛[9C
```
 What is a bit strange is that the GT911 driver seams to initialize like:
 
 DRVR: GT911 Started (480x272)
 
 Installed is the the version for 800x480 pixels as stated above.

**Reply by 001oc** [reply]
I have tried the display.ini from tasmota/displaydesc folder too but that is even not working:

```
:H,ILI9341,800,480,16,SPI,1,*,*,*,*,*,*,*,40
:S,2,1,1,0,40,20
:I
EF,3,03,80,02
CF,3,00,C1,30
ED,4,64,03,12,81
E8,3,85,00,78
CB,5,39,2C,00,34,02
F7,1,20
EA,2,00,00
C0,1,23
C1,1,10
C5,2,3e,28
C7,1,86
36,1,48
37,1,00
3A,1,55
B1,2,00,18
B6,3,08,82,27
F2,1,00
26,1,01
E0,0F,0F,31,2B,0C,0E,08,4E,F1,37,07,10,03,0E,09,00
E1,0F,00,0E,14,03,11,07,31,C1,48,08,0F,0C,31,36,0F
11,80
29,80
:o,28
:O,29
:A,2A,2B,2C,16
:R,36
:0,48,00,00,00
:1,28,00,00,01
:2,88,00,00,02
:3,E8,00,00,03
:i,20,21
:TI1,5d,*,*,-1,38
#

```

**Reply by sfromis** [reply]
You can find the autoconf content in directories like this:
https://github.com/tasmota/autoconf/tree/main/raw/esp32s3/SunTon-S3-Display-5inch
As you can see, the display.ini file there says ST7262, but that is really just a name string without effect on device operation. However, the [template page](https://templates.blakadder.com/sunton_ESP32-8048S050.html) for Sunton 5" is for ESP32-8048S050, which may be a rather different board from yours.

You never posted a link to a sales page for your **exact** board, which might have possible further hints, and it is pretty important whether it is RGB or SPI interface; these are totally different. With the many wires in the main ribbon cable to the display, I really do not believe that it is SPI. Other than your experimentation with a openHASP binary, do you have anything to suggest that it is ILI9341?

If you instead manually put the files into the ESP32 file system, you can edit all you want. Beware that you should not have both autoconf and extracted files at the same time, to avoid any ambiguity about what is being used.

I'd think that the openHASP boot log reporting detection of GT911 should be pretty reliable, but not really anything in other details of the hardware. While it may seem odd that the GT911 message says (480x272), there is no need for touch to have the same resolution as the display itself, as single pixels are hard to touch anyway....

**Reply by 001oc** [reply]
@sfromis you are right, sorry for that, here we go, that is the link I used to buy my board:

https://www.aliexpress.us/item/3256805758992113.html?spm=a2g0o.order_list.order_list_main.17.6f8d5c5fQ0VDAG&gatewayAdapt=glo2usa4itemAdapt

I guess it will be a rgb board. But as aliexpress description are in general not very reliable I am not sure at all. And of course I don't have any reliable hint which kind of board, display, touch I have but the GT911 chip at the cable of the touchscreen.

Concerning the both autoconfig and display.ini files I even thought that it wouldn't make sence to have maybe conflicting values in differnt files. As of that I have deleted the autoconfig file and stayed with the display.ini for now.

Concerning the openHASP bootlog I agree that touch could be identified correctly but there is unfortunately no information about the display itself.

Right now I dont understand much concerning tasmota and displays in general. I am used to modify and compile my tasmota images by myself (VSCode/platformio). I have alot tasmota running in my (smart)home but none with displays till now. I  use mqtt and http mostly to do controlling stuff. Displays, touch,  LVGL, HASPmota and so on in connection to tasmota is new to me.

I think the pcb of my board is a 2 layer one. I will try to find out which line is connecting the display to which pin of the ESP. Maybe that could help to find out more.

**Reply by 001oc** [reply]
this one looks like my display:

https://de.aliexpress.com/item/1005006030813472.html?src=google&aff_fcid=db3158915923436eb87bfdfb8f777b70-1708415667712-01808-UneMJZVf&aff_fsk=UneMJZVf&aff_platform=aaf&sk=UneMJZVf&aff_trace_key=db3158915923436eb87bfdfb8f777b70-1708415667712-01808-UneMJZVf&terminal_id=a89f525dc561423ab555a10ec70fb045&afSmartRedirect=y&gatewayAdapt=glo2deu

the cable looks very similar, maybe identical. They sell it as a **4,3 Zoll tft lcd Display 800*480 ips RGB st7262 lcd Modul**

**Reply by 001oc** [reply]
I tried this as display.ini but no effect:

:H,ST7262,800,480,16,RGB,40,41,39,42,2,15,16,4,45,48,47,21,14,8,3,46,9,1,5,6,7,12
:S,2,1,1,0,40,20
:V,0,8,4,8,0,8,4,8,1
:0,00
:1,01
:2,02
:3,03
:TI1,5d,*,*,-1,38
#

**Reply by sfromis** [reply]
There is a display descriptor for ST7262 RGB, but specifics may not be the same as your board.
https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/ST7262_rgb16_display.ini
However, it is very similar to the Sunton 5" autoconf I found earlier
https://github.com/tasmota/autoconf/blob/main/raw/esp32s3/SunTon-S3-Display-5inch/display.ini

**Reply by sfromis** [reply]
It is also important what specific Tasmota template you're using, and whether you make sure to have the power toggle (front page or command) on.

**Reply by 001oc** [reply]
I use the following template currently:

{"NAME":"SUNTON","GPIO":[1,1,1,1,1,1,1,1,1,1,6720,704,736,672,1,1,1,1,1,640,608,1,0,0,0,0,0,6210,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":1}

**Reply by sfromis** [reply]
When booting on a similar 7" Sunton display, I'm getting boot time messages like this, at log level 3. You should basically get very similar messages, just with a different display name. The line about SPI initialization is not so important, as my display is also a RGB display, instead of SPI. Other than that, deviations may be a clue to "something missing"....
```
00:00:00.001 HDW: ESP32-S3 v0.2 (PSRAM)
00:00:00.040 UFS: FlashFS mounted with 12564 kB free
00:00:00.050 CFG: Loaded from File, Count 148
00:00:00.056 QPC: Count 1
00:00:00.059 SPI: Hardware using GPIO12(CLK), GPIO11(MOSI) and GPIO13(MISO)
00:00:00.059 I2C: Bus1 using GPIO20(SCL) and GPIO19(SDA)
00:00:00.664 TFS: File 'mcp23x.dat' not found
00:00:00.665 MCP: Invalid template
00:00:00.665 TYA: Active=0
00:00:00.667 ROT: Mode 1
00:00:00.733 BRY: Berry initialized, RAM used 6016 bytes
00:00:00.753 BRY: No 'preinit.be'
00:00:00.768 DSP: File descriptor used
00:00:00.769 I2C: GT911 found at 0x5d
00:00:00.888 UTI: GT911 initialized
00:00:01.395 DSP: Configured display 'EK9716'
00:00:01.400 Project tasmota - xxxxxxxx Version 13.4.0.1(lvgl-haspmota)-2_0_14(2024-02-15T10:02:41)
...
00:00:01.411 ETH: No ETH MDC (SPI CS), ETH MDIO (SPI IRQ) and ETH POWER (SPI RST) GPIO defined
00:00:01.433 SHT: Sensor did not ACK command
00:00:01.769 LVG: LVGL initialized
00:00:02.067 BRY: Successfully loaded 'autoexec.be'
00:00:02.100 LVG: Refreshed 16000 pixels in 25 ms (640 pix/ms)
00:00:02.129 LVG: Refreshed 16000 pixels in 24 ms (666 pix/ms)
...
```
This is when also having a `pages.jsonl` setup causing something to be displayed right away, this simple color gradient test setup:
```
{"page":0}
{"id":101,"obj":"btn","x":  8,"y":8,"w":48,"h":24,"action":"prev","bg_color":"#1fa3ec","radius":10,"border_side":1,"text":"\uF053","text_font":"montserrat-20"}
{"id":102,"obj":"btn","x": 88,"y":8,"w":48,"h":24,"action":"back","bg_color":"#1fa3ec","radius":10,"border_side":1,"text":"\uF015","text_font":"montserrat-20"}
{"id":103,"obj":"btn","x":168,"y":8,"w":48,"h":24,"action":"next","bg_color":"#1fa3ec","radius":10,"border_side":1,"text":"\uF054","text_font":"montserrat-20"}
{"page":1}
{"id":0,"bg_color":"#000000","bg_grad_color":"#ff0000","bg_grad_dir":1,"text_color":"#FFFFFF"}
{"id":1,"obj":"label","x":32,"y":48,"w":140,"text":"Red","align":0,"text_font":"montserrat-20"}
{"page":2}
{"id":0,"bg_color":"#000000","bg_grad_color":"#00ff00","bg_grad_dir":1,"text_color":"#FFFFFF"}
{"id":1,"obj":"label","x":32,"y":48,"w":140,"text":"Green","align":0,"text_font":"montserrat-20"}
{"page":3}
{"id":0,"bg_color":"#000000","bg_grad_color":"#0000ff","bg_grad_dir":1,"text_color":"#FFFFFF"}
{"id":1,"obj":"label","x":32,"y":48,"w":140,"text":"Blue","align":0,"text_font":"montserrat-20"}
```
